src/server/modes/base/maintenance/spectating.ts

```diff
@@ -138,16 +139,29 @@ export default class GameSpectating extends System {
 
       viewport.subs.clear();
 
+      /**
+       * Choose a random player to spectate
+       */
+      const playerId =
+        this.playerIds.length > 0 ? this.playerIds[getRandomInt(0, this.playerIds.length - 1)] : 0;
+
+      spectator.spectate.current = playerId;
+      this.subscribeToViewport(playerId, spectatorId);
+
```

Comment:
I'm not sure about this behavior regarding FFA and CTF. I remember for sure that in CTF after switching to spectate mode, I could stay to observe the area of the map I was on at the moment of switching to spectate mode without auto assignment the player to spectate.

But I would like someone else to confirm it, then I would add a condition only for BTR.

Reply:
I do not know what the behaviour was of CTF, but I think on FFA it would pick a random player.

I'll analyse this from the proxy logs and let you know what I find.

Reply:
Now I think it's better to leave your implementation. This switching logic has several advantages: it reduces the number of objects in the collision system, and it complicates the use of timers in CTF a bit.

Also I'm going to add the feature proposed by xplay: in CTF to switch to the player who currently has taken the flag, i.e. to sort `playerIds` depending on the game state and mode.

I suppose the same adapts for FFA: to switch to the player who was the last killer of the switcher, and to sort other players by the bounty value.

Reply:
Agreed that feature will be an improvement over the current behaviour. As the implementation of the original game servers will not be copied, I won't continue with the proxy logs analysis.